### PR TITLE
Make native query editor read only if user doesn't have write permissions

### LINKED ISSUE #3514: If I don't have SQL writing permission, I should't see the "open editor" button
This is the button in question. We should hide this if the user doesn't have SQL editing/writing perms for the given DB.

screen shot 2016-10-11 at 2 37 03 pm

### PR DESCRIPTION
**Pull request by tlrobinson**
Resolves #3514

### CODE REVIEW

File: frontend/src/metabase/query_builder/NativeQueryEditor.jsx [hunk 1]
```diff
@@ -96,12 +99,15 @@ export default class NativeQueryEditor extends Component {
...
-        if (this.state.modeInfo && editor.getSession().$modeId !== this.state.modeInfo.mode) {
-            console.log('Setting ACE Editor mode to:', this.state.modeInfo.mode);
-            editor.getSession().setMode(this.state.modeInfo.mode);
-            // monkey patch the mode to add our bracket/paren/braces-matching behavior
-            if (!editor.getSession().$mode.$behaviour) {
-                editor.getSession().$mode.$behaviour = new SQLBehaviour();
+        if (modeInfo) {
+            editor.setReadOnly(modeInfo.database.native_permissions !== "write");
+            if (editor.getSession().$modeId !== modeInfo.mode) {
+                console.log('Setting ACE Editor mode to:', modeInfo.mode);
```
Comment (kdoh) [inline frontend/src/metabase/query_builder/NativeQueryEditor.jsx]: Do we want to keep this logging in non dev?

Comment (tlrobinson) [inline frontend/src/metabase/query_builder/NativeQueryEditor.jsx]: Removed

**Comment by salsakran** [comment]
This works but is a little weird to use, as there's no visual indicator of the editor input being read-only. It seems like it's broken. Can we prevent the cursor in the input element?

**Comment by mazameli** [comment]
What Sameer proposes would be helpful. Alternatively, the `show query` button could just open the same modal that the `view sql` button does, but with a close button instead of a `convert to sql` button:

screen shot 2016-10-13 at 3 21 56 pm

If we go with that behavior, I'd change the icon next to the `show query` button.

**Comment by salsakran** [comment]
*strong* -1 for a modal here.

**Comment by mazameli** [comment]
Duh, yeah, if you need to be looking at the query and its results at the same time, it's a no-go.

**Comment by tlrobinson** [comment]
I've hidden the cursor in read-only mode.

**Comment by salsakran** [comment]
something went weird for me on safari..

screen shot 2016-10-17 at 2 19 11 pm

**Comment by camsaul** [comment]
@salsakran you need to specify the collection to run queries against if you're using Mongo

**Comment by salsakran** [comment]
ahhh... nvm then. I got a little confused, but it makes sense. 

This feels way better.

**Pull request merged**